### PR TITLE
Slash Command Implementation and Refactoring into Files. 

### PR DESCRIPTION
**Pull request by A-Phamfam**
### Node 16.9 or greater
* discord.js requires Node 16.9 or greater to operate. Make sure to update node when executing code.

### Environment Variables

* `.env.example` file was created and `README.md` file was updated to include instructions for each environment variable.
* The following environment variables were added: `GUILD_ID`, `BOT_ID`, `HELP_CHANNEL_IDS`, and `HELP_CHANNEL_NAMES`
* Removed help_channel_ids and help_channel_names from source code and moved to `HELP_CHANNEL_IDS` and `HELP_CHANNEL_NAMES` for easier developer testing.
* The `GUILD_ID` environment variable should be defined for fast registration and immediate running of commands. This is the "test" guild which commands are immediately registered to. Due to slash command registration, if the test guild is not listed here it will take about 1 hour for Discord to register the command globally. 

### ESLint

* Added ESLint to enforce styling rules. `npx eslint --fix ./` can be run to automatically fix styling errors or the IDE or editor plugin can be used.

### Abstracting out of index.js to `./src` files
* Extracted most of the code out of `index.js` into more modular files. The following were files originally in `index.js` but are now independent files that are imported where needed.
  * `./src/cronjobs.js`
  * `./src/download.js`
  * `./src/minesweeper.js`
  * `./src/render.js`
  * `./src/router.js`
  * `./src/templates.js`
* Deleted `roll.js` because I felt it was kind of unnecessary lol.
### Event abstraction
* Events that were done by `client.on('eventName'')` are now imported in from `./events`. It is now easier to directly add or change the behavior of specific bot events from these event files.

### why
* Didn't realize that `!why` was a meme so wasted time refactoring it lol. You can revert that commit if you want. Commit SHA: `f4a4af50339163a16596097d2d5d6651177d0601`

## Slash Commands
* The old Regex commands are still kept inside `./src/router.js`. Some of them were removed since they are no longer compatible with the new DiscordAPI, but all the important ones are still there.
* Bot commands are now created in the `./commands` directory with each separate file representing a different command. These mostly all behave the same as the old Regex commands and are mostly copy and pasted with some editing to make it compatible with the new abstractions.
* Instructions on how to create new commands were added to the `README.md`
* Autocomplete was added for the **docs** command. In order to provide the autocomplete options, a new function `docAutocomplete(query)` was added to `./src/mathworks-docs.js` which does similar requests as `searchDocs(query)` but formats the object to be compatible with the DiscordAPI.

## Notes about Slash Commands update
* Make sure the bot has the **MESSAGE CONTENT INTENT** enabled in the Discord Developer Portal.
* The `discord.js` library has made major changes to the fields and methods of different objects, so updating to the newest `discord.js` version will make it incompatible with the old Matlab bot code.

### CODE REVIEW

**Comment by MujyKun** [comment]
good luck to whoever is reviewing this PR.

**Review by FannoFlow — APPROVED** [review]
Ship it and see what breaks

File: src/inchat-octave.js [hunk 1]
```diff
@@ -1,30 +1,235 @@
...
+
+    // Check for illegal command usage and warn against it...
+    const found_illegal_match = code.match(illegal_use_regexp);
+    if (found_illegal_match) {
+        msg.channel.send(`Someone was being naughty <@${process.env.OWNER_ID}>`);
+        return;
+    }
+
+    // Write the users code command to a file. continue execution if it works
+    fs.writeFile(ic_octave_user_code, code, (err) => {
```
Comment (Xetera) [inline src/inchat-octave.js:39]: Shouldn't these files be cleaned up after execution 👀

File: src/cronjobs.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    }, {
+        name: 'Youtube',
+        use: getNewestVideo,
+        interval: 2 * 3600 * 1e3,
+        template: 'youtube.md',
+        errors: [],
+    },
+];
+
+const cronjob_data = JSON.parse(fs.readFileSync(cronjob_data_file, 'utf8'));
```
Comment (Xetera) [inline src/cronjobs.js:30]: JSON files can be required directly without reading the filesystem fyi

**Review by Xetera — COMMENTED** [review]
gaming

File: src/inchat-octave.js (hunk 1 above)
Comment (arzulo) [reply src/inchat-octave.js:39]: yes! Thanks for pointing out

File: src/cronjobs.js (hunk 2 above)
Comment (arzulo) [reply src/cronjobs.js:30]: Good point, thanks!

**Review by arzulo — APPROVED** [review]
LGTM

**Pull request merged**